src/lfortran/semantics/ast_common_visitor.h

```diff
@@ -2860,183 +2860,119 @@ class CommonVisitor : public AST::BaseVisitor<Derived> {
+                                        bool is_convertible = false;
+                                        ASR::ArrayConstant_t *a = ASR::down_cast<ASR::ArrayConstant_t>(cast->m_arg);
+                                        ASR::ttype_t* cast_type = cast->m_type;
+                                        Vec<ASR::expr_t*> body;
+                                        body.reserve(al, ASRUtils::get_fixed_size_of_array(a->m_type));
                                         if (cast_kind == ASR::cast_kindType::IntegerToReal) {
-                                            bool is_int = true;
-                                            ASR::ArrayConstant_t *a = ASR::down_cast<ASR::ArrayConstant_t>(cast->m_arg);
-                                            ASR::ttype_t* real_type = cast->m_type;
-                                            Vec<ASR::expr_t*> body;
-                                            body.reserve(al, ASRUtils::get_fixed_size_of_array(a->m_type));
                                             for (size_t i = 0; i < (size_t) ASRUtils::get_fixed_size_of_array(a->m_type); i++) {
                                                 ASR::expr_t *e = ASRUtils::fetch_ArrayConstant_value(al, a, i);
-                                                // it will be IntegerConstant_t convert it to RealConstant_t
                                                 if (ASR::is_a<ASR::IntegerConstant_t>(*e)) {
                                                     ASR::IntegerConstant_t *int_const = ASR::down_cast<ASR::IntegerConstant_t>(e);
                                                     double val = int_const->m_n;
                                                     ASR::expr_t *real_const = ASRUtils::EXPR(ASR::make_RealConstant_t(al, int_const->base.base.loc,
-                                                        val, ASRUtils::type_get_past_array(real_type)));
+                                                        val, ASRUtils::type_get_past_array(cast_type)));
                                                     body.push_back(al, real_const);
-                                                } else {
-                                                    is_int = false;
-                                                    break;
-                                                }
-                                            }
-                                            if (is_int) {
-                                                ASR::expr_t* array_const = ASRUtils::EXPR(ASRUtils::make_ArrayConstructor_t_util(al, a->base.base.loc, body.p, body.size(), real_type, a->m_storage_format));
-                                                cast->m_value = ASRUtils::expr_value(array_const);
-                                                value = cast->m_value;
+                                                    is_convertible = true;
+                                                } else break;
```

Comment:
This looks strange. Let's use the regular `else` with `{ }` as used at other places.

```suggestion
                                                } else { 
                                                    break;
                                                }
```

Similarly for other `else break;` in this PR.